CMakeLists.txt

```
@@ -282,7 +282,11 @@ if(NOT MSVC)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O0 -g")
     set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -G -g")
   else()
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3")
+    if (DEFINED EMSCRIPTEN)
+      # for some reason -O3 is causing functions to not be exported in Emscripten
+    else()
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3")
+    endif()
```

```suggestion
    if (NOT DEFINED EMSCRIPTEN)
      # for some reason -O3 is causing functions to not be exported in Emscripten
      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3")
    endif()
```

This should be equivalent and doesn't leave an empty control body.